blueprints/controllers/ikea_e1524_e1810/ikea_e1524_e1810.yaml

```
@@ -31,12 +31,24 @@ blueprint:
             - deCONZ
             - ZHA
             - Zigbee2MQTT
+            - Zigbee2MQTT_V2
     controller_device:
-      name: (deCONZ, ZHA) Controller Device
-      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA.
+      name: (deCONZ, ZHA, Zigbee2MQTT_V2) Controller Device
+      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA, Zigbee2MQTT_V2.
```

```suggestion
      name: (deCONZ, ZHA, Zigbee2MQTT) Controller Device
      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA, Zigbee2MQTT.
```

Instead of considering Z2M v2.0.0 as a separate integration, I'd consider it as the actual "Zigbee2MQTT" integration without distinctions. To check whether legacy triggers want to be used, we can simply check whether the user provided a `controller_entity`.

To me it sounds like this approach would make it easier to drop support for legacy triggers in the future - just remove the `controller_entity` input + any business logic related to it.

---

I think that is a more user-friendly way of doing it and aides in tech debt for future versions.

@yarafie would you know where to go with this suggestion, if you would like to take it up?
Would @EPMatt be able to support if needed on this?

---

Just some observations.
I tried rewriting my philips controller to this (as that I could test and agree that it's more user friendly).

The new triggers they require a valid device id, which is only the case if a Controller Device is selected.
So when the blueprint is updated the automations becomes unavailable until a controller device has been selected.

So I don't think it's possible to avoid a breaking change unless choosing another route.
I added my changes here [https://github.com/Nicolai-/awesome-ha-blueprints/tree/ns_mqtt_device_triggers](https://github.com/Nicolai-/awesome-ha-blueprints/tree/ns_mqtt_device_triggers)

There is another way to integrate and that is the MQTT Trigger which can listen on a [mqtt topic](https://www.home-assistant.io/docs/automation/trigger/#mqtt-trigger).
The main drawback from that is that the topic should be entered manually by the user
Atleast I haven't found a way to fetch it from a device/entity.